exercises/word-count/test/test_word_count.c

```diff
@@ -0,0 +1,428 @@
+#include <stddef.h>
+#include <stdlib.h>
+#include <string.h>
+#include "vendor/unity.h"
+#include "../src/word_count.h"
+
+WordCount_Word_t ActualSolution[MAX_WORDS];
+
+static void test_solution(
+   WordCount_Word_t *expected_solution,
+   int expected_word_count,
+   WordCount_Word_t *actual_solution,
+   int actual_word_count)
+{
+   int index;
+
+   // All words counted?
+   TEST_ASSERT_EQUAL(expected_word_count, actual_word_count);
+
+   // now test the word count for the words...
+   for (index=0; index < actual_word_count; index++)
+   {
+      TEST_ASSERT_EQUAL(expected_solution[index].count, actual_solution[index].count);
+      TEST_ASSERT_EQUAL_STRING(expected_solution[index].text, actual_solution[index].text);
+   }
+}
+
+static void reset_actual_solution(void)
+{
+   memset(ActualSolution, 0, sizeof(ActualSolution));
+}
+
+// Tests Start HERE
+void test_word_count_one_word(void)
+{
+   int index = 0;
+   int actual_word_count;
+   char *input_text = "word";
+   const int expected_word_count = 1;
+   WordCount_Word_t ExpectedSolution[expected_word_count];
+
+   // build the expected solution
+   memset(ExpectedSolution, 0, sizeof(ExpectedSolution));   // clear to start with a known value
+
+   // fill in the words!
+   ExpectedSolution[index].count = 1;
+   strncpy(ExpectedSolution[index++].text, "word", MAX_WORD_LENGTH);
+
+   reset_actual_solution();
+
+   // call the implementation routine!
+   actual_word_count = word_count(input_text, ActualSolution);
+
+   test_solution(
+      ExpectedSolution,
+      expected_word_count,
+      ActualSolution,
+      actual_word_count);
+}
+
+void test_word_count_one_of_each_word(void)
+{
+   int index = 0;
+   int actual_word_count;
+   char *input_text = "one of each";
+   const int expected_word_count = 3;
+   WordCount_Word_t ExpectedSolution[expected_word_count];
+
+   // build the expected solution
+   memset(ExpectedSolution, 0, sizeof(ExpectedSolution));   // clear to start with a known value
+
+   // fill in the words! - hardcode but easiest way to init?
```

Comment:
I think it's fine to fill in the expected words in this manner.  You could possibly add a helper function like `AddToExpectedWords()` or something like that.